### DISCUSSION TITLE
Is there a way to make the user verify the new email before updating their Ory identity?

### DISCUSSION THREAD

**Discussion by alevosia** [ory/kratos#3576] [Q&A]
Hi.

We are currently migrating from Auth0 to Ory.

When a user that has signed up and is verified updates their email address, is there a way to make the user verify their email address first before the email of their Ory identity is updated?

Currently, when completing a Settings Flow, the user's email is immediately updated even if they haven't verified it. This can cause users losing access to their account if the new email address they have entered is wrong.

**Comment by vinckr** [comment]
Hello @alevosia 

I think the way to solve this would be to [require a verified email address](https://console.ory.sh/projects/current/verification) on login. 
So the user needs to verify their email before they can log in. 
This would I think be the easiest solution. 

Another solution would be using a custom UI for the settings flow, but then it would still be possible to be changed with unverified email through the API - maybe not an issue for this use case.
